kbi/0011/index.rst

```diff
@@ -0,0 +1,236 @@
+.. index::
+   single: installation; datalad
+   single: installation; extensions
+
+KBI0011: Debugging installation issues related to multiple Python versions
+==========================================================================
+
+:authors: Michał Szczepanik <m.szczepanik@fz-juelich.de>
+:discussion: https://github.com/psychoinformatics-de/knowledge-base/pull/43
+:keywords: python version, package manager, pip, conda, MacOS, linux, Windows
+:software-versions: datalad_0.18.3, python_3.9.2
+
+It is not uncommon to have multiple Python versions installed on a
+single machine. It can be needed, for example, when using several
+software stacks with different requirements (some programs can even
+come with their own Python installation). `Virtual environments`_ and
+`conda environments`_ are two popular methods of managing multiple
+versions of Python packages and / or Python itself.
+
+DataLad is a Python package which also provides a command line
+interface (``datalad`` command). DataLad can be installed with
+different methods, including package managers such as ``apt``,
+``homebrew``, ``conda``, ``pip``, or others, depending on user
+preferences. DataLad extensions are also distributed through different
+channels, but are most commonly installed using ``pip``, the Package
+Installer for Python.
+
+Like with any other command line tool, the actual executable used when
+calling the ``datalad`` command is determined by searching several
+locations in the order determined by the `PATH variable`_. If DataLad
+is installed in several locations (e.g. different Python versions or
+multiple virtual environments), the installation used in a given
+moment will be dependent on current ``PATH`` (affected by system
+settings, user settings, and potentially modified by an active virtual
+environment).
+
+In certain situations, the resolution order may lead to unexpected
+DataLad configurations, for example when the ``datalad`` command is
+provided in one Python installation, but ``pip`` comes from another,
+or if an older Python version takes precedence in ``PATH``. In the
+latter case, a seemingly fresh DataLad install may provide an outdated
+version (DataLad roughly follows the `Python release cycle`_; e.g.,
+support for Python 3.6 has been dropped in DataLad 0.16).
+
+This KBI presents examples of two such installation issues, and
+discusses simple and useful debugging steps.
+
+.. _virtual environments: https://docs.python.org/3/library/venv.html
+.. _conda environments: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html
+.. _PATH variable: https://en.wikipedia.org/wiki/PATH_(variable)
+.. _Python release cycle: https://devguide.python.org/versions/
+
+Examples
+--------
+
+Example 1
+^^^^^^^^^
```

Comment:
```suggestion
Conda competing with Homebrew on MacOS
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I think it makes sense to allow a reader to decide if they are interested in a particular example

Reply:
Absolutely! Thanks for suggesting the section titles - my creativity on that day has ended while writing :)